presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -387,6 +423,146 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+                    break;
+                }
+                else {
+                    throw e;
+                }
+            }
+        }
+
+        SchemaTableName viewName = viewMetadata.getTable();
+
+        Table.Builder tableBuilder = Table.builder()
+                .setDatabaseName(viewName.getSchemaName())
+                .setTableName(viewName.getTableName())
+                .setOwner(session.getUser())
+                .setTableType(VIRTUAL_VIEW)
+                .setDataColumns(columns)
+                .setPartitionColumns(ImmutableList.of())
+                .setParameters(properties)
+                .setViewOriginalText(Optional.of(encodeViewData(viewData)))
+                .setViewExpandedText(Optional.of(PRESTO_VIEW_EXPANDED_TEXT_MARKER));
+
+        tableBuilder.getStorageBuilder()
+                .setStorageFormat(VIEW_STORAGE_FORMAT)
+                .setLocation("");
+        Table table = tableBuilder.build();
+        PrincipalPrivileges principalPrivileges = buildInitialPrivilegeSet(session.getUser());
+
+        Optional<Table> existing = metastore.getTable(metastoreContext, viewName.getSchemaName(), viewName.getTableName());
+        if (existing.isPresent()) {
+            if (!replace || !isPrestoView(existing.get())) {
+                throw new ViewAlreadyExistsException(viewName);
+            }
+
+            metastore.replaceTable(metastoreContext, viewName.getSchemaName(), viewName.getTableName(), table, principalPrivileges);
+            return;
+        }
+
+        try {
+            metastore.createTable(metastoreContext, table, principalPrivileges);
+        }
+        catch (TableAlreadyExistsException e) {
+            throw new ViewAlreadyExistsException(e.getTableName());
+        }
+    }
+
+    private MetastoreContext getMetastoreContext(ConnectorSession session)
+    {
+        return new MetastoreContext(session.getIdentity(), session.getQueryId(), session.getClientInfo(), session.getSource(), getMetastoreHeaders(session), isUserDefinedTypeEncodingEnabled(session), HiveColumnConverterProvider.DEFAULT_COLUMN_CONVERTER_PROVIDER);
+    }
+
+    public List<SchemaTableName> listViews(ConnectorSession session, Optional<String> schemaName)
+    {
+        ImmutableList.Builder<SchemaTableName> tableNames = ImmutableList.builder();
+        MetastoreContext metastoreContext = getMetastoreContext(session);
+        for (String schema : listSchemas(session, schemaName.orElse(null))) {
+            for (String tableName : metastore.getAllViews(metastoreContext, schema).orElse(Collections.emptyList())) {
+                tableNames.add(new SchemaTableName(schema, tableName));
+            }
+        }
+        return tableNames.build();
+    }
+
+    public Map<SchemaTableName, ConnectorViewDefinition> getViews(ConnectorSession session, SchemaTablePrefix prefix)
+    {
+        ImmutableMap.Builder<SchemaTableName, ConnectorViewDefinition> views = ImmutableMap.builder();
```

Comment:
This method is also 99% the same as the one in HiveMetadata. Please extract common util methods

Reply:
Done.